tests/SpecTests/Operation.php

```diff
@@ -214,6 +239,40 @@ private function execute(Context $context)
         }
     }
 
+    /**
+     * Executes the client operation and return its result.
+     *
+     * @param Client  $client
+     * @param Context $context Execution context
+     * @return mixed
+     * @throws LogicException if the collection operation is unsupported
+     */
+    private function executeForClient(Client $client, Context $context)
+    {
+        $args = $context->prepareOptions($this->arguments);
+        $context->replaceArgumentSessionPlaceholder($args);
+
+        switch ($this->name) {
+            case 'listDatabaseNames':
+                return array_map(
+                    function (DatabaseInfo $info) {
+                        return $info->getName();
+                    },
+                    iterator_to_array($client->listDatabases())
+                );
+
+            case 'listDatabaseObjects':
```

Comment:
We have a few options to skip these tests:
* dropping the respective files from the spec tests folder
* skipping them via the `skipReason` field in the spec file itself.
* skipping them in the spec test itself if the test name begins with one of the skipped operation names

Of all these, the last option is the most resilient when we sync spec tests in the future, as it allows us to copy test files over from the specifications directory without having to worry about changing them or skipping some of them. I agree that we may as well skip them since the entire logic for these operations is contained in the tests.

> So good on you for going the extra mile :)

Credit should go do the C driver team, which [also implemented this the same way](https://github.com/mongodb/mongo-c-driver/blob/b37f263e2286ad20c656df1e96fa106377fd7c7a/src/libmongoc/tests/json-test-operations.c#L1909). I wasn't sure how to implement the operation, so I looked there for guidance. This also highlights that their implementation has the same flaw as mine, so we may want to also skip these tests there or implement the operation correctly.